### ISSUE TITLE
Unable to add attachment when using brand new accesstoken

### ISSUE THREAD

**Issue by gemiware** [justmiles/go-markdown2confluence#55]
See https://github.com/justmiles/go-confluence/issues/3

When using the brandspanking new  accesstoken functionality we are able to create pages on our corporate confluence using SSO. So that is great! However when we also have attachments on the page it fails with a 401 response statuscode.

Here is the debug output
```
DEBU[0000] GET https://confluence-url.com/rest/api/content/352026659/child/attachment?filename=DataModel.png  
DEBU[0000] Response Status Code: 200                    
DEBU[0000] Response Body: '{"results":[{"id":"352026661","type":"attachment","status":"current","title":"DataModel.png","metadata":{"mediaType":"image/png","labels":{"results":[],"start":0,"limit":200,"size":0,"_links":{"self":"https://confluence-test.topaas.ns.nl/rest/api/content/352026661/label"}},"_expandable":{"currentuser":"","properties":"","frontend":"","editorHtml":""}},"extensions":{"mediaType":"image/png","fileSize":104340,"comment":""},"_links":{"webui":"/display/GPLLS/leverschema-datamodel?preview=%2F352026659%2F352026661%2FDataModel.png","download":"/download/attachments/352026659/DataModel.png?version=1&modificationDate=1667228691590&api=v2","thumbnail":"/download/thumbnails/352026659/DataModel.png?api=v2","self":"https://confluence-test.topaas.ns.nl/rest/api/content/352026661"},"_expandable":{"container":"/rest/api/content/352026659","operations":"","children":"/rest/api/content/352026661/child","restrictions":"/rest/api/content/352026661/restriction/byOperation","history":"/rest/api/content/352026661/history","ancestors":"","body":"","version":"","descendants":"/rest/api/content/352026661/descendant","space":"/rest/api/space/GPLLS"}}],"start":0,"limit":50,"size":1,"_links":{"self":"https://confluence-test.topaas.ns.nl/rest/api/content/352026659/child/attachment?filename=DataModel.png","base":"https://confluence-url.com","context":""}}' 
DEBU[0000] POST https://confluence-url.com/rest/api/content/352026659/child/attachment/352026661/data 
DEBU[0000] Response Status Code: 401                    
DEBU[0000] Response Body: ''                            
[unexpected end of JSON input]

```

When I try the same using postman, it does work, so it is not a confluence rights issue.

Could it be the POST for the attachment data uses the requestWithFunc function from the confluence client? That only has basic auth at the moment...

See https://github.com/justmiles/go-confluence/blob/83494e1d603019bfab31d9d9f9ddfa2b210f575f/client.go#L116

The normal request function has logic for adding the access-token to the request, the requestWithFunc function does not.

**Comment by justmiles** [comment]
@gemiware, appreciate you taking the time to work through this! I don't have access to a corporate instance so your feedback is very helpful.


This is fixed in the latest version - [v3.4.1](https://github.com/justmiles/go-markdown2confluence/releases/tag/v3.4.1). If you could confirm or provide feedback against your corporate instance for this release I'd really appreciate it.

Thanks again!

**Comment by gemiware** [comment]
v3.4.1 fixes the attachment issue! Thanks again.

```
DEBU[0000] POST https://confluence-url.com/rest/api/content/352026683/child/attachment 
DEBU[0001] Response Status Code: 200         
DEBU[0001] Response Body: '{"results":[{"id":"352026684","type":"attachment","status":"current","title":"DataModel.png","version":{"by":{"type":"known","username":"daUser","userKey":"40280ed383eb4a430183f535ac2b002d","profilePicture":{"path":"/images/icons/profilepics/default.svg","width":48,"height":48,"isDefault":true},"displayName":"Da User Write","_links":{"self":"https://confluence-url.com/rest/api/user?key=40280ed383eb4a430183f535ac2b002d"},"_expandable":{"status":""}},"when":"2022-11-01T15:53:54.299+01:00","message":"d41d8cd98f00b204e9800998ecf8427e","number":1,"minorEdit":false,"hidden":false,"_links":{"self":"https://confluence-orl.com/rest/experimental/content/352026684/version/1"},"_expandable":{"content":"/rest/api/content/352026684"}},"container":{"id":"352026683","type":"page","status":"current","title":"datamodel","extensions":{"position":"none"},"_links":{"webui":"/display/GPLLS/datamodel","edit":"/pages/resumedraft.action?draftId=352026683","tinyui":"/x/O4D7F","self":"https://confluence-url.com/rest/api/content/352026683"},"_expandable":{"container":"/rest/api/space/GPLLS","metadata":"","operations":"","children":"/rest/api/content/352026683/child","restrictions":"/rest/api/content/352026683/restriction/byOperation","history":"/rest/api/content/352026683/history","ancestors":"","body":"","version":"","descendants":"/rest/api/content/352026683/descendant","space":"/rest/api/space/GPLLS"}},"metadata":{"comment":"d41d8cd98f00b204e9800998ecf8427e","mediaType":"image/png","_expandable":{"currentuser":"","properties":"","frontend":"","editorHtml":""}},"extensions":{"mediaType":"image/png","fileSize":104340,"comment":"d41d8cd98f00b204e9800998ecf8427e"},"_links":{"webui":"/display/GPLLS/datamodel?preview=%2F352026683%2F352026684%2FDataModel.png","download":"/download/attachments/352026683/DataModel.png?version=1&modificationDate=1667314434299&api=v2","thumbnail":"/download/thumbnails/352026683/DataModel.png?api=v2","self":"https://confluence-url.com/rest/api/content/352026684"},"_expandable":{"operations":"","children":"/rest/api/content/352026684/child","restrictions":"/rest/api/content/352026684/restriction/byOperation","history":"/rest/api/content/352026684/history","ancestors":"","body":"","descendants":"/rest/api/content/352026684/descendant","space":"/rest/api/space/GPLLS"}}],"size":1,"_links":{"base":"https://confluence-url.com","context":""}}' 

```
